### ISSUE TITLE
RSCD step error for some MIRI exposures

### ISSUE THREAD

**Issue by stscijgbot-jp** [spacetelescope/jwst#4928]
_Issue [JP-1446](https://jira.stsci.edu/browse/JP-1446) was created on JIRA by [Howard Bushouse](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bushouse):_

Some MIRI imaging exposures are throwing the following error from the RSCD step (perhaps confined to exposures with READPATT='FASTGRPAVG'):
```java
2020-05-11 14:17:02,278 - CRDS - DEBUG -  {'META.EXPOSURE.READPATT [READPATT]': 'FASTGRPAVG',
2020-05-11 14:17:02,278 - CRDS - DEBUG -  'META.EXPOSURE.TYPE [EXP_TYPE]': 'MIR_IMAGE',
2020-05-11 14:17:02,278 - CRDS - DEBUG -  'META.INSTRUMENT.BAND [BAND]': 'UNDEFINED',
2020-05-11 14:17:02,278 - CRDS - DEBUG -  'META.INSTRUMENT.CHANNEL [CHANNEL]': 'UNDEFINED',
2020-05-11 14:17:02,278 - CRDS - DEBUG -  'META.INSTRUMENT.CORONAGRAPH [CORONMSK]': 'UNDEFINED',
2020-05-11 14:17:02,278 - CRDS - DEBUG -  'META.INSTRUMENT.DETECTOR [DETECTOR]': 'MIRIMAGE',
2020-05-11 14:17:02,278 - CRDS - DEBUG -  'META.INSTRUMENT.FILTER [FILTER]': 'F2100W',
2020-05-11 14:17:02,278 - CRDS - DEBUG -  'META.INSTRUMENT.LAMP_STATE [LAMP]': 'OFF',
2020-05-11 14:17:02,278 - CRDS - DEBUG -  'META.INSTRUMENT.NAME [INSTRUME]': 'MIRI',
2020-05-11 14:17:02,278 - CRDS - DEBUG -  'META.OBSERVATION.DATE [DATE-OBS]': '2016-01-17',
2020-05-11 14:17:02,278 - CRDS - DEBUG -  'META.OBSERVATION.TIME [TIME-OBS]': '01:48:20.253',
2020-05-11 14:17:02,278 - CRDS - DEBUG -  'META.SUBARRAY.NAME [SUBARRAY]': 'FULL',
2020-05-11 14:17:02,278 - CRDS - DEBUG -  'META.VISIT.TSOVISIT [TSOVISIT]': 'F',
2020-05-11 14:17:02,278 - CRDS - DEBUG -  'REFTYPE': 'UNDEFINED'}
2020-05-11 14:17:02,279 - CRDS - DEBUG -  Reference type 'rscd' defined as 'jwst_miri_rscd_0010.fits'
2020-05-11 14:17:02,280 - stpipe.Detector1Pipeline.rscd - INFO - Using RSCD reference file /ifs/archive/test/jwst/ref/tmp_crds/cache/references/jwst/miri/jwst_miri_rscd_0010.fits
Traceback (most recent call last):
  File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-0.16.0.20200507-py37/bin/strun", line 26, in <module>
    step = Step.from_cmdline(sys.argv[1:])
  File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-0.16.0.20200507-py37/lib/python3.7/site-packages/jwst/stpipe/step.py", line 156, in from_cmdline
    return cmdline.step_from_cmdline(args)
  File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-0.16.0.20200507-py37/lib/python3.7/site-packages/jwst/stpipe/cmdline.py", line 346, in step_from_cmdline
    step.run(*positional)
  File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-0.16.0.20200507-py37/lib/python3.7/site-packages/jwst/stpipe/step.py", line 392, in run
    step_result = self.process(*args)
  File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-0.16.0.20200507-py37/lib/python3.7/site-packages/jwst/pipeline/calwebb_detector1.py", line 86, in process
    result = self.rscd(result)
  File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-0.16.0.20200507-py37/lib/python3.7/site-packages/jwst/stpipe/step.py", line 392, in run
    step_result = self.process(*args)
  File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-0.16.0.20200507-py37/lib/python3.7/site-packages/jwst/rscd/rscd_step.py", line 52, in process
    result = rscd_sub.do_correction(input_model, rscd_model, self.type)
  File "/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-0.16.0.20200507-py37/lib/python3.7/site-packages/jwst/rscd/rscd_sub.py", line 48, in do_correction
    group_skip = param['skip']
KeyError: 'skip'
----------------------------------------------------------------------
ERROR RUNNING STEP 'Detector1Pipeline':
    'skip'
----------------------------------------------------------------------
```

**Comment by karllark** [comment]
@jemorrison 
I agree.  At this point the RSCD algorithm does not handle groups that have been averaged.  Might be possible to update it to do so, but for now skipping it is the right answer.

**Comment by stscijgbot-jp** [comment]
_Comment by [Howard Bushouse](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bushouse) on [JIRA](https://jira.stsci.edu/browse/JP-1446?focusedCommentId=463487#comment-463487):_

I suggest putting this logic into the rscd_step module itself, rather than in the calwebb_detector1 pipeline. The step can always be called, but it can then just issue a warning that it's skipping for an exposure with this READPATT.

**Comment by hbushouse** [comment]
Fixed by #4934

**Comment by stscijgbot-jp** [comment]
_Comment by [Howard Bushouse](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bushouse) on [JIRA](https://jira.stsci.edu/browse/JP-1446?focusedCommentId=463687#comment-463687):_

Fixed by #4934
